File: src/main/java/io/github/gravetii/scene/help/WhatIsItComponent.java
package io.github.gravetii.scene.help;

import io.github.gravetii.controller.BaseController;
import io.github.gravetii.scene.FxComponent;

import javafx.scene.control.TextArea;

public class WhatIsItComponent extends FxComponent<BaseController, TextArea> {
  protected WhatIsItComponent() throws Exception {
    super("whatIsIt.fxml");
    this.create();
  }

  @Override
  protected BaseController createController() {
    return new BaseController();
  }

  @Override
  protected TextArea createNode() throws Exception {
    return this.loadNode();
  }
}

File: src/main/kotlin/io/github/gravetii/scene/help/WhatIsItComponent.kt
package io.github.gravetii.scene.help

import io.github.gravetii.controller.BaseController
import io.github.gravetii.scene.FxComponent
import javafx.scene.control.TextArea

class WhatIsItComponent : FxComponent<BaseController, TextArea>("whatIsIt.fxml") {

    override val controller: BaseController = BaseController()

    override val node: TextArea = loadNode()

}